fdbserver/TLogServer.actor.cpp

```diff
@@ -1784,9 +1784,11 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 
 	state Version poppedVer = poppedVersion(logData, reqTag);
 
+	auto tagData = logData->getTagData(reqTag);
+	bool tagRecovered = tagData && !tagData->unpoppedRecovered;
```

Comment:
Should this be `tagRecovered = !tagData || tagData->unpoppedRecovered`. I.e., no tag data means tag has recovered.

Reply:
no, `tagData->unpoppedRecovered` means there is data that has not yet been popped, and the log has not yet recovered.